### PR TITLE
fix(meta): use per table mce and safe epoch in metadata backup

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

After #17161, batch query over metadata backup won't work correctly, because it `try_get_hummock_version` (find a queryable backup) using global safe_epoch and MCE, while `validate_safe_epoch` using per table safe_epoch and MCE.

This PR fixes it by storing and using per table safe_epoch and MCE.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by zwang28** [comment]
FYI, I have to revert the adoption of the macro introduced in #17225 when restoring a backup, because insertion requires a specific order to satisfy foreign key constraints. @yezizp2012

File: src/storage/backup/src/lib.rs [hunk 1]
```diff
@@ -58,6 +60,8 @@ pub struct MetaSnapshotMetadata {
     #[serde(default)]
     pub format_version: u32,
     pub remarks: Option<String>,
+    #[serde(with = "table_id_key_map")]
```
Comment (wenym1) [inline src/storage/backup/src/lib.rs:63]: If we set key type as string or u32, we may not need to impl the serde by ourselves?

File: src/meta/src/backup_restore/restore_impl/v2.rs [hunk 2]
```diff
@@ -93,28 +93,40 @@ impl WriterModelV2ToMetaStoreV2 {
...
-
-risingwave_backup::for_all_metadata_models_v2!(define_write_model_v2_to_meta_store_v2);
-
 #[async_trait::async_trait]
 impl Writer<MetadataV2> for WriterModelV2ToMetaStoreV2 {
     async fn write(&self, target_snapshot: MetaSnapshot<MetadataV2>) -> BackupResult<()> {
         let metadata = target_snapshot.metadata;
         let db = &self.meta_store.conn;
-        write_model_v2_to_meta_store_v2(&metadata, db).await?;
+        insert_models(metadata.seaql_migrations.clone(), db).await?;
```
Comment (wenym1) [inline src/meta/src/backup_restore/restore_impl/v2.rs:101]: Any reason to manually expand the macro here?

**Review by wenym1 — COMMENTED** [review]
Rest LGTM

File: src/meta/src/backup_restore/restore_impl/v2.rs (hunk 2 above)
Comment (wenym1) [reply src/meta/src/backup_restore/restore_impl/v2.rs:101]: Oh I saw the comment now.

File: src/storage/backup/src/lib.rs (hunk 1 above)
Comment (zwang28) [reply src/storage/backup/src/lib.rs:63]: serde_json requires map key to be String. So either using String as key or impl the serde.

The downside of using String as key is additional type conversion when visiting the map.

**Review by wenym1 — APPROVED** [review]
LGTM

**Pull request merged**